File: app/src/main/java/com/mytlogos/enterprise/background/room/EditDao.java
package com.mytlogos.enterprise.background.room;

import androidx.room.Dao;
import androidx.room.Query;

import com.mytlogos.enterprise.background.room.model.RoomEditEvent;

import java.util.List;

@Dao
public interface EditDao extends MultiBaseDao<RoomEditEvent> {
    @Query("SELECT * FROM RoomEditEvent")
    List<RoomEditEvent> getAll();

    @Query(
            "SELECT * FROM RoomEditEvent " +
                    "WHERE (:id <= 0 OR id=:id) " +
                    "AND (:objectType <= 0 OR objectType=:objectType)" +
                    "AND (:eventType <= 0 OR eventType=:eventType)"
    )
    List<RoomEditEvent> getAll(int id, int objectType, int eventType);
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/EditDao.kt
package com.mytlogos.enterprise.background.room

import androidx.room.Dao
import androidx.room.Query
import com.mytlogos.enterprise.background.room.model.RoomEditEvent

@Dao
interface EditDao : MultiBaseDao<RoomEditEvent?> {
    @get:Query("SELECT * FROM RoomEditEvent")
    val all: MutableList<RoomEditEvent>

    @Query("""SELECT * FROM RoomEditEvent
WHERE (:id <= 0 OR id=:id)
AND (:objectType <= 0 OR objectType=:objectType)
AND (:eventType <= 0 OR eventType=:eventType)""")
    fun getAll(id: Int, objectType: Int, eventType: Int): List<RoomEditEvent>
}